pkg/ingestor/parser/slsa/parser_slsa.go

```diff
@@ -68,10 +70,28 @@ func (s *slsaParser) Parse(ctx context.Context, doc *processor.Document) error {
 func (s *slsaParser) getSubject(statement *in_toto.ProvenanceStatement) {
 	// append artifact node for the subjects
 	for _, sub := range statement.Subject {
+		var nodes []assembler.ArtifactNode
 		for alg, ds := range sub.Digest {
-			s.subjects = append(s.subjects, assembler.ArtifactNode{
+			nodes = append(nodes, assembler.ArtifactNode{
 				Name: sub.Name, Digest: alg + ":" + strings.Trim(ds, "'"), NodeData: *assembler.NewObjectMetadata(s.doc.SourceInformation)})
 		}
+		s.subjects = append(s.subjects, nodes...)
+
+		// Add heuristic to link an artifact to a package if the subject name starts with "pkg:"
+		if strings.HasPrefix(sub.Name, "pkg:") {
```

Comment:
Unless the subject is specially crafted I don't imagine it would start with `pkg:`. What are we imagining the use case would be here?

I'm thinking we could improve the heuristics here so that we can match container image artifacts imported from SLSA and packages imported from sboms by

1. normalizing names for images that are on docker Hub (e.g. ianmlewis/demo vs. docker.io/ianmlewis/demo vs. index.docker.io/ianmlewis/demo)
2. Linking package and artifact by name and digest for the top-level SBOM package.